src/client/components/equalizer/equalizer.js

```diff
@@ -76,158 +54,153 @@ class Equalizer extends Component {
       navigator.mediaDevices.getUserMedia({
         audio: true,
       })
-        .then(stream => this.createSoundStream(stream))
-        .catch((e) => {
-          throw new Error(e);
+        .then(
+          this.createSoundStream(),
+        )
+        .catch((errorStream) => {
+          throw new Error(errorStream);
         });
     } else {
       throw new Error('browser doesnt support audio API');
     }
   }
 
-  widthMerge = (eventFromInputTypeRange) => {
-    const { mergeCanvasWidth: mergeCanvasWidthAsProp } = this.props;
-    mergeCanvasWidthAsProp(eventFromInputTypeRange);
-  }
-
-  playSoundFromFile = () => {
-    const { audioData } = this.props;
-    const { audioFromFile: soundFromFile, playPauseState } = audioData;
-    if (!playPauseState) {
-      soundFromFile.play();
-      this.equaliserRun();
-    } else {
-      soundFromFile.pause();
-    }
-    const { playPauseSoundFromFile: playPauseSoundFromFileAsProp } = this.props;
-    playPauseSoundFromFileAsProp();
+  uploadSoundInfoFromFile = (e) => {
+    const [file] = e.target.files;
+    const audioFile = new Audio(URL.createObjectURL(file));
+    const sound = new Pizzicato.Sound({
+      source: 'file',
+      options: {
+        path: audioFile.src,
+        loop: true,
+      },
+    }, () => this.createSoundInfoInState(sound, file));
+  };
+
+  createSoundInfoInState = (sound, file) => {
+    const { audioData: { analyser } } = this.props;
+    sound.connect(analyser);
+    const { createAudioDataAsProp } = this.props;
```

Comment:
If you already destruct smth from props, no need to did it again, just destruct all in one place

Reply:
ok